lib/useButter.js

```diff
@@ -0,0 +1,225 @@
+     */
+    const {
+        cancelRequest,
+        controller: fetchController,
+    } = useCancelFetch(friendlyType);
+
+
+    /**
+     * Fetches data from the ButterCMS API at the specified endpoint using provided parameters.
+     * 
+     * @async
+     * @function get
+     * @param {string} [apiEndpoint=BASE_PATH] - The API endpoint to send the request to.
+     * @param {Object} [localParams={}] - Additional parameters to include in the request.
+     * @returns {Promise<Object>} An object containing the data, meta, and any errors from the API response.
+     * @throws Will throw an error object containing errors, config, and params if the API response includes errors.
+     */
+    async function get (apiEndpoint = BASE_PATH, localParams = {}) {
+        // Extract apiToken and separate it from the rest of the user's configuration
+        const {
+            apiToken: auth_token,
+            ...userConfig
+        } = butterConfig;
+        
+        // Initialize headers for the request
+        const butterHeaders = new Headers({
+            "Content-Type": "application/json",
+            "X-Butter-Client": `JS/${ BUTTER_CLIENT_VERSION }`
+        });
+        
+        // Append additional header for server-side environments
+        if (typeof window === "undefined") {
+            butterHeaders.append("Accept-Encoding", "gzip");
+        }
+        
+        // Prepare the request configuration using the onRequest hook
+        const {
+            config,
+            headers,
+            params
+        } = await useOnRequest(
+            apiEndpoint,
+            auth_token,
+            {
+                cancelRequest,
+                config: userConfig,
+                headers: butterHeaders,
+                params: localParams,
+                type,
+            }
+        );
+        
+        try {
+            // Set up fetch timeout and signal
+            const {
+                cleanup,
+                signal
+            } = useFetchTimeout(fetchController, friendlyType, config.timeout);
+
+            // Perform the API request
+            const response = await fetch(
+                `${ apiEndpoint }?${ new URLSearchParams(params) }`,
+                {
+                    cache: config.cache,
+                    method: "GET",
+                    headers,
+                    signal
+                }
+            );
+            
+            // Clean up after the fetch timeout
+            cleanup();
+            
+            // Process the response using the onResponse hook
+            const {
+                data,
+                meta,
+                ...errors
+            } = await useOnResponse(
+                response,
+                {
+                    config,
+                    // pass the request as user won't see our
+                    // changes for auth_token or testMode
+                    // remove testMode params and auth_token in onResponse
```

Comment:
I'm not sure I understand this comment 100%

Reply:
Right, so `testMode` is a config option that the dev sets up. But it controls two params (`test` and `preview`), this is a behind the scenes sort of thing and we just want the dev-controlled params to be shown rather than the ones that the API changes. Same goes for the auth_token. Its added as a param by our system, but the dev doesn't add it to params themselves. We want a clean dev-defined params to be send to the hook.